--- wireshark-1.99.1/epan/wslua/lua_bitop.c.omv~	2015-07-22 23:13:04.737195945 +0200
+++ wireshark-1.99.1/epan/wslua/lua_bitop.c	2015-07-22 23:13:18.405229968 +0200
@@ -74,7 +74,7 @@ static UBits barg(lua_State *L, int idx)
 #endif
 #elif defined(LUA_NUMBER_INT) || defined(LUA_NUMBER_LONG) || \
       defined(LUA_NUMBER_LONGLONG) || defined(LUA_NUMBER_LONG_LONG) || \
-      defined(LUA_NUMBER_LLONG)
+      defined(LUA_NUMBER_LLONG) || defined(LUA_INTEGER)
   if (sizeof(UBits) == sizeof(lua_Number))
     b = bn.b;
   else
--- wireshark-1.99.7/acinclude.m4.omv~	2015-07-22 23:26:18.839902767 +0200
+++ wireshark-1.99.7/acinclude.m4	2015-07-22 23:26:37.366938648 +0200
@@ -870,23 +870,16 @@ AC_DEFUN([AC_WIRESHARK_LIBLUA_CHECK],[
 			AC_MSG_CHECKING(the Lua version)
 			lua_ver=`$AWK AS_ESCAPE('/LUA_VERSION_NUM/ { print $NF; }' $header_dir/lua.h | sed 's/0/./')`
 
-			if test "x$lua_ver" = "x5.3"
-			then
-				# Wireshark doesn't compile with Lua 5.3 today
-				AC_MSG_RESULT($lua_ver - disabling Lua support)
-				have_lua=no
-			else
-				AC_MSG_RESULT($lua_ver)
+			AC_MSG_RESULT($lua_ver)
 
-				wireshark_save_CPPFLAGS="$CPPFLAGS"
-				CPPFLAGS="$CPPFLAGS -I$header_dir"
-				AC_CHECK_HEADERS(lua.h lualib.h lauxlib.h, ,
-				[
-					have_lua=no
-					# Restore our CPPFLAGS
-					CPPFLAGS="$wireshark_save_CPPFLAGS"
-				])
-			fi
+			wireshark_save_CPPFLAGS="$CPPFLAGS"
+			CPPFLAGS="$CPPFLAGS -I$header_dir"
+			AC_CHECK_HEADERS(lua.h lualib.h lauxlib.h, ,
+			[
+				have_lua=no
+				# Restore our CPPFLAGS
+				CPPFLAGS="$wireshark_save_CPPFLAGS"
+			])
 
 			if test "x$have_lua" = "x"
 			then
