diff -up wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/about_dialog.cpp.1~ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/about_dialog.cpp
--- wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/about_dialog.cpp.1~	2024-07-17 22:58:48.000000000 +0200
+++ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/about_dialog.cpp	2024-07-18 15:11:08.188797605 +0200
@@ -491,12 +491,12 @@ void AboutDialog::updateWiresharkText()
      * conversions from QByteArray to QString at -O2 and higher due to
      * computing a branch that will never be taken.
      */
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_OFF(stringop-overread)
 #endif
     clipboardInfo += gstring_free_to_qbytearray(get_compiled_version_info(gather_wireshark_qt_compiled_info)) + "\n";
     clipboardInfo += gstring_free_to_qbytearray(get_runtime_version_info(gather_wireshark_runtime_info)) + "\n";
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_ON(stringop-overread)
 #endif
 }
diff -up wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/follow_stream_dialog.cpp.1~ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/follow_stream_dialog.cpp
--- wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/follow_stream_dialog.cpp.1~	2024-07-17 22:58:48.000000000 +0200
+++ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/follow_stream_dialog.cpp	2024-07-18 15:11:08.188797605 +0200
@@ -914,11 +914,11 @@ void FollowStreamDialog::showBuffer(QByt
              * conversions from QByteArray to QString at -O2 and higher due to
              * computing a branch that will never be taken.
              */
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_OFF(stringop-overread)
 #endif
             yaml_text += "      " + base64_data.toBase64() + "\n";
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_ON(stringop-overread)
 #endif
 
diff -up wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/interface_toolbar.cpp.1~ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/interface_toolbar.cpp
--- wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/interface_toolbar.cpp.1~	2024-07-17 22:58:48.000000000 +0200
+++ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/interface_toolbar.cpp	2024-07-18 15:11:08.188797605 +0200
@@ -13,14 +13,14 @@
 
 #include <ws_diag_control.h>
 
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_OFF(stringop-overflow)
 #if WS_IS_AT_LEAST_GNUC_VERSION(13,0)
 DIAG_OFF(restrict)
 #endif
 #endif
 #include "interface_toolbar.h"
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_ON(stringop-overflow)
 #if WS_IS_AT_LEAST_GNUC_VERSION(13,0)
 DIAG_ON(restrict)
diff -up wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/show_packet_bytes_dialog.cpp.1~ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/show_packet_bytes_dialog.cpp
--- wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/show_packet_bytes_dialog.cpp.1~	2024-07-17 22:58:48.000000000 +0200
+++ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/show_packet_bytes_dialog.cpp	2024-07-18 15:11:08.188797605 +0200
@@ -896,11 +896,11 @@ void ShowPacketBytesDialog::updatePacket
              * conversions from QByteArray to QString at -O2 and higher due to
              * computing a branch that will never be taken.
              */
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_OFF(stringop-overread)
 #endif
             text.append("  " + base64_data.toBase64() + "\n");
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_ON(stringop-overread)
 #endif
         }
diff -up wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/utils/data_printer.cpp.1~ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/utils/data_printer.cpp
--- wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/utils/data_printer.cpp.1~	2024-07-18 15:11:08.188797605 +0200
+++ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/utils/data_printer.cpp	2024-07-18 15:12:05.901286453 +0200
@@ -104,11 +104,11 @@ void DataPrinter::toClipboard(DataPrinte
         break;
     case DP_UTF8Text:
         // This automatically compensates for invalid UTF-8 in the input
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_OFF(stringop-overread)
 #endif
         clipboard_text += QString::fromUtf8(printData);
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_ON(stringop-overread)
 #endif
         break;
diff -up wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/wireshark_main_window_slots.cpp.1~ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/wireshark_main_window_slots.cpp
--- wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/wireshark_main_window_slots.cpp.1~	2024-07-17 22:58:48.000000000 +0200
+++ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/wireshark_main_window_slots.cpp	2024-07-18 15:11:08.192797639 +0200
@@ -947,11 +947,11 @@ void WiresharkMainWindow::startCapture(Q
                      * even though the isEmpty() check guarantees that
                      * filter_ba.constData() is never NULL or empty.
                      */
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_OFF(stringop-overread)
 #endif
                     if (strcmp(interface_opts->cfilter, filter_ba.constData()) != 0) {
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_ON(stringop-overread)
 #endif
                         /* No, so not all selected interfaces have the same capture
diff -up wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/utils/data_printer.cpp.omv~ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/utils/data_printer.cpp
--- wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/utils/data_printer.cpp.omv~	2024-07-18 15:26:21.113899294 +0200
+++ wireshark-4.3.0-build/wireshark-4.3.0/ui/qt/utils/data_printer.cpp	2024-07-18 15:26:34.506150004 +0200
@@ -122,11 +122,11 @@ DIAG_ON(stringop-overread)
         }
         break;
     case DP_Base64:
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_OFF(stringop-overread)
 #endif
         clipboard_text = printData.toBase64();
-#if WS_IS_AT_LEAST_GNUC_VERSION(12,1)
+#if WS_IS_AT_LEAST_GNUC_VERSION(12,1) && !defined(__clang__)
 DIAG_ON(stringop-overread)
 #endif
         break;
