--- wireshark-1.99.7/configure.ac.p1~	2015-07-22 23:23:16.770544171 +0200
+++ wireshark-1.99.7/configure.ac	2015-07-22 23:23:49.690609906 +0200
@@ -3011,7 +3011,7 @@ dnl check whether plugins should be enab
 dnl check for plugins directory - stolen from Amanda's configure.ac
 dnl
 dnl we don't wish to expand ${libdir} yet
-plugindir='${libdir}/wireshark/plugins/${VERSION}'
+plugindir='${libdir}/wireshark'
 AC_ARG_WITH(plugins,
   AC_HELP_STRING( [--with-plugins@<:@=DIR@:>@],
 		  [support plugins (installed in DIR, if supplied) @<:@default=yes, if possible@:>@]),
