--- wireshark-1.10.0/configure.ac.orig	2013-06-15 17:31:57.400334988 +0200
+++ wireshark-1.10.0/configure.ac	2013-06-15 17:32:43.564757854 +0200
@@ -2232,7 +2232,7 @@ dnl check whether plugins should be enab
 dnl check for plugins directory - stolen from Amanda's configure.ac
 dnl
 dnl we don't wish to expand ${libdir} yet
-plugindir='${libdir}/wireshark/plugins/${VERSION}'
+plugindir='${libdir}/wireshark'
 AC_ARG_WITH(plugins,
   AC_HELP_STRING( [--with-plugins@<:@=DIR@:>@],
                   [support plugins (installed in DIR, if supplied) @<:@default=yes, if possible@:>@]),
