diff -up wireshark-1.8.4/configure.in.automake-1_13 wireshark-1.8.4/configure.in
--- wireshark-1.8.4/configure.in.automake-1_13	2013-01-03 01:04:32.260229773 +0100
+++ wireshark-1.8.4/configure.in	2013-01-03 01:05:48.804488167 +0100
@@ -1960,7 +1960,7 @@ ifdef(_CUSTOM_AC_OUTPUT_,, define(_CUSTO
 sinclude(asn1/Custom.m4) dnl
 ifdef(_CUSTOM_ASN1_AC_OUTPUT_,, define(_CUSTOM_ASN1_AC_OUTPUT_, )) dnl
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_OUTPUT(
   Makefile
   doxygen.cfg
