--- configure.in	2007-12-19 00:52:27.000000000 +0100
+++ configure.in.oden	2007-12-19 00:53:54.000000000 +0100
@@ -50,7 +50,7 @@
 	AC_PATH_PROG(HTML_VIEWER, htmlview)
 	if test "x$HTML_VIEWER" = x
 	then
-		AC_DEFINE_UNQUOTED(HTML_VIEWER, "mozilla", [HTML viewer, e.g. mozilla])
+		AC_DEFINE_UNQUOTED(HTML_VIEWER, "www-browser", [HTML viewer, e.g. mozilla])
 	else
 		AC_DEFINE_UNQUOTED(HTML_VIEWER, "htmlview", [HTML viewer, e.g. mozilla])
 	fi
