--- wireshark-1.10.14/epan/dissectors/packet-canopen.c.orig	2015-06-27 07:39:16.026751595 +0100
+++ wireshark-1.10.14/epan/dissectors/packet-canopen.c	2015-06-27 07:39:31.123752728 +0100
@@ -272,7 +272,7 @@
 
         /* add CANopen frame type */
 
-        type_ti = proto_tree_add_text(canopen_tree, tvb, tvb_reported_length(tvb),
+        type_ti = proto_tree_add_text(canopen_tree, tvb, 0, tvb_reported_length(tvb),
                                       "Type: %s", function_code_str);
         canopen_type_tree = proto_item_add_subtree(type_ti, ett_canopen);
 
